docs/Trust Registry.md

```diff
@@ -56,3 +57,136 @@ where `"z5Bug71M7Sj7cYpbVBDmN:2:test_schema:0.3"` represents the schema ID, name
 > **_NOTE_**: In a production environment, this should not be exposed to the internet or interacted with directly.
 > It's advisable to either avoid exposing this to the internet or set up a separate security layer for the trust
 > registry. This is because it's crucial to prevent unauthorized individuals from making changes to the trust registry.
+
+## Trust-registry Role in application flows
+
+Below we indicate where and how the Trust-registry is consulted to verify that Issuers/Verifiers and Schemas are on the
+Trust-registry.
+
+### Create Credential Definition
+
+```mermaid
+---
+title: Trust-registry called during credential definition creation
+---
+flowchart LR
+    App(Create Credential Definition Request) -->|Consults| TR[Trust Registry]
+    subgraph Trust Registry Checks
+      TR -->|Validates| Check1{Issuer Authorization}
+      Check1 -->|If Unauthorized| Block[⨯ Block Operation]
+      Check1 -->|If Authorized| Check2{Schema exists on TR}
+      Check2 -->|Not on TR| Block
+    end
+    Check2 -->|If Registered| Continue[✓ Proceed with Creation]
+    
+    style TR fill:#a8d1ff,stroke:#1e88e5,color:black
+    style Block fill:#ffcdd2,stroke:#e53935,color:black
+    style Continue fill:#c8e6c9,stroke:#43a047,color:black
+```
+
+---
+
+### Credential issuance
+
+```mermaid
+---
+title: Create Credential-Offer/Sending Credential
+---
+flowchart LR
+    subgraph Request Types
+        App1(Create Offer <br> Connectionless) --> Consults[Consults]
+        App2(Send Credential <br> with Connection ID) --> Consults[Consults]
+    end
+    subgraph Trust Registry Checks
+      Consults --> TR
+      TR[Trust Registry] -- Validates --> Check1{Issuer<br>Authorization}
+      Check1 -->|If Unauthorized| Block[⨯ Block Operation]
+      Check1 -->|If Authorized| Check2{Schema<br>Registration}
+      Check2 -->|Not on TR| Block
+    end
+    Check2 -->|If Registered| Proceed[Continue]
+    
+
+    subgraph Request Continue
+      Proceed -->|Connectionless| Continue1[✓ Create Offer]
+      Proceed -->|With Connection ID| Continue2[✓ Send Credential]
+    end
+
+    linkStyle 0,8 stroke:#ff7043,color:#ff7043, stroke-width:2
+    linkStyle 1,9 stroke:#7cb342,color:#7cb342,stroke-width:2
+    
+    style TR fill:#a8d1ff,stroke:#1e88e5,color:black
+    style Block fill:#ffcdd2,stroke:#e53935,color:black
+    style Continue1 fill:#c8e6c9,stroke:#43a047,color:black
+    style Continue2 fill:#c8e6c9,stroke:#43a047,color:black
+    style Proceed fill:#c8e6c9,stroke:#43a047,color:black
+
+```
```

Comment:
Let's please add some text with each diagram. The text should be a summary of what the diagram is showing. 
e.g., when sending a credential, we validate that the issuing party has the required issuer role on the trust registry. The schema associated with the credential must also be on the trust registry.
Simple summary for each diagram please